smos-report/src/Smos/Report/OptParse/Types.hs

```diff
@@ -0,0 +1,52 @@
+module Smos.Report.OptParse.Types where
+
+import Import
+
+import Data.Configurator.Types
+import qualified Data.Text as T
+
+data ShouldPrint
+    = PrintError
+    | PrintWarning
+    | DontPrint
+    deriving (Show, Eq)
+
+parseShouldPrint :: String -> Maybe ShouldPrint
+parseShouldPrint "error" = Just PrintError
+parseShouldPrint "warning" = Just PrintWarning
+parseShouldPrint "nothing" = Just DontPrint
+parseShouldPrint _ = Nothing
```

Comment:
These two lines are a bit confusing.
I would imagine that `nothing` would be parsed as `Nothing`.
Not a big deal.